django/contrib/auth/hashers.py

```diff
@@ -416,26 +446,36 @@ def encode(self, password, salt):
         data = bcrypt.hashpw(password, salt)
         return "%s$%s" % (self.algorithm, data.decode('ascii'))
 
+    def decode(self, encoded):
+        algorithm, empty, algostr, work_factor, data = encoded.split('$', 4)
+        assert algorithm == self.algorithm
+        salt, checksum = data[:22], data[22:]
+        return {
+            'algorithm': algorithm,
+            'algostr': algostr,
+            'work_factor': work_factor,
```

Comment:
Add `int` around work_factor here?

Reply:
Personally I think the function should return whatever the underlying datatype is. As per my other comment I can make the `work_factor` data type an int. That said, I had been aiming to keep all the data types in the dict consistent.

Reply:
Yes please, for further reusability of the `decode` function it makes sense to use `int` where the underlying data is actually an `int`. As for consistency that imo went away once we switched `iterations` to `int`. And I still think this switch makes sense since we do not have to call `int` all over the place where we use it. If you think of our endgoal (ie something like a `salt_len` as result of `decode`) so we can update `must_update` to account for the salt, it makes even more sense to have integers. Imo the key->value relation is well defined and with python being as dynamic as it is, let's make use of that.

Reply:
> As for consistency that imo went away once we switched iterations to int.

lol. That it did.

> And I still think this switch makes sense...

That's what matters to me. :)

> Imo the key->value relation is well defined and with python being as dynamic as it is, let's make use of that.

I view the issue more in terms of coming to grasps with a new code base and I see simple typing as a benefit to a new reader of the code. That said, I'm not a maintainer of django so I see my preferences as accounting for nothing :)